index.js

```diff
@@ -39,25 +33,37 @@ var lnCms = angular.module('lnCms', [
             name: route.state,
             url: route.url,
             templateUrl: 'templates/' + route.template + '/template.html',
+            controller: 'LnViewController as vm',
+            resolve: {
+              staticData: ['lnCmsClientService', function(lnCmsClientService){
+                return lnCmsClientService.getStatic();
+              }],
+              viewData: ['lnCmsClientService', '$stateParams', function(lnCmsClientService, $stateParams){
+                var endpoint = route.endpoint || 'post';
+                var params = route.params || {};
+                angular.merge(params, $stateParams);
+                return lnCmsClientService.getData(endpoint, params);
+              }],
+            },
             data: {
               endpoint: (route.endpoint || 'post'),
               fixedParams: (route.params || {}),
               request: 'request' in route ? route.request : true
             }
```

Comment:
Mmm, just thinking about this again.. Maybe it would be useful to have this data attached to the state, just in case we have to call the state endpoint directly from the app (?).